City-of-Helsinki/helfi-etusivu-news-search
# UHF-5914 Filters

# [UHF-5914 Filters](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5914)

## What was done?

* Fixed issues with webpack / hds
* Added general component for dropdowns
* Added filters defined in the sketch
* Added executing query when pressing submit button
* Some general tweaks and utility hooks

## How to install
* Check install instructions for Drupal here: https://github.com/City-of-Helsinki/drupal-helfi-etusivu/pull/106
* ... OR: `nvm use; npm i; npm start` for the standalone app

## How to test
* Check that app looks like sketch (https://share.goabstract.com/8be8c801-48b8-40b5-866e-ec197360fccf?collectionLayerId=d66cb334-acef-41a8-a6c2-86432f251928&mode=design&sha=latest)
* Search functionality things to test for:
  * Filters components should work
  * Filters and results should consider the language context
  * Query should be executed only when pressing the submit button OR the 'clear all' button
  * No results should render 'no results' text
  * Changing the filter values should be represented in the url query string

## Designer check

- [ ] This branch has been checked by the designer
